packages/block-library/src/search/edit.native.js

```diff
@@ -52,6 +54,11 @@ export default function SearchEdit( {
 	const [ isPlaceholderSelected, setIsPlaceholderSelected ] = useState(
 		true
 	);
+	const [ isLongButton, setIsLongButton ] = useState( false );
+	const [ containerWidth, setContainerWidth ] = useState(
```

Comment:
I think here we could use the prop `blockWidth` that every block receives in the `props` from the parent wrapper. This way we can only have just one `onLayout` for the button. What do you think?

You have it available [here](https://github.com/WordPress/gutenberg/blob/e4a5461b98b6cb9da62f8a4add4e42a3cce59a09/packages/block-library/src/search/edit.native.js#L45).

Reply:
Something related to `Dimensions` to take into account is that depending on where/how we use it we might need to add a listener to dimensions changes, an example: iPad in split view, if you resize the space you might end up with the first value you got from the Dimensions helper.

I _think_ you won't need this if you use the `blockWidth` prop because it will trigger a re-render from the parent [here](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/mobile/readable-content-view/index.native.js).

There's `useWindowDimensions` from React native but there's a bug in the version we have now, once we update we could use that 🤞

Reply:
Awesome, thanks for the tips @geriux , I was wondering if there was a way to get the `blockWidth` so this is really helpful.

Reply:
Updated in 0fcfd43fe01f3edfcd12cd3694950fdbf820ffcd